src/argowrapper/engine/argo_engine.py

```diff
@@ -582,59 +585,69 @@ def get_workflow_logs(self, workflow_name: str, uid: str) -> List[Dict]:
             )
 
     def workflow_submission(self, request_body: Dict, auth_header: str):
+        # Lock function so only one can run at a time
+        self.lock.acquire()
```

Comment:
is it possible to lock based on a username, so that a single user can only submit one workflow at a time, but different users can still submit in parallel?

Reply:
code updated